src/shamir/mod.rs

```diff
@@ -0,0 +1,226 @@
+//! This module implements Shamir's Secret Sharing Scheme.
+//!
+//! Given a secret represented as a u128, the secret is split into a set
+//! of shares such that any subset of shares of size at least `threshold` (k,n) where n = 2k-1
+//! can reconstruct the secret using Lagrange interpolation over a finite field.
+//!
+//! The chosen finite field is GF(PRIME) with PRIME = 2^127 - 1. Be sure that the provided
+//! secret is less than PRIME.
+
+use rand::Rng;
+
+/// Prime modulus for finite field arithmetic: 2^127 - 1.
+const PRIME: u128 = 170141183460469231731687303715884105727;
```

Comment:
why have we chosen a specific prime, and not made it using ronkathon field implementation?